### PR TITLE
Remove 'undefined' street addresses

### PR DESCRIPTION
**Pull request by CloCkWeRX**
Fix https://github.com/alltheplaces/alltheplaces/issues/8830

### CODE REVIEW

**Comment by CloCkWeRX** [comment]
@matkoniecz fyi. If you have a wider list; would be good to figure out if these are all storerocket

**Comment by matkoniecz** [comment]
eg_america_us.geojson has `"addr:street_address": "Undefined"`

teriyaki_madness_us.geojson has `"addr:street_address": "undefined"`

circle_k.geojson has `"addr:street_address": "UNDEFINED"`

gant.geojson has `"phone": "undefined"`

sparkasse_de.geojson has `"phone": "+undefinedundefinedundefined"`

neuhaus.geojson has `"phone": "undefined0227848669"`

starbucks_eu.geojson has `"addr:full": "undefined, BA11 4QE Frome", "addr:street_address": "undefined"`

fastrac_cafe_us.geojson has `"addr:street_address": "Undefined"`

orlen.geojson has `"phone": "---"`

**Comment by matkoniecz** [comment]
Oh, and also `"addr:housenumber": "b/n"` in poczta_polska_pl.geojson

orlen_paczka_pl.geojson has `"addr:street_address": "Genera\u0142a W\u0142adys\u0142awa Sikorskieg B/N"` but it is more annoying to fix if even wrong

groszek_pl.geojson with `"addr:housenumber": "b/n"`

some occasional b/n indicating missing numbers scattered across other addresses in Polish spiders

church_of_england_gb.geojson with single `"name": "------"`

**Comment by matkoniecz** [comment]
"b/n" is annoying as I bet that in some place they managed to make it valid housenumber, so probably should be removed per spider, for Polish ones... Especially as it short form of Polish text, so unlikely to be used outside Poland.

**Comment by CloCkWeRX** [comment]
Okay, adjusted the PR so this should be cleaned up:

eg_america_us.geojson has "addr:street_address": "Undefined"
teriyaki_madness_us.geojson has "addr:street_address": "undefined"
circle_k.geojson has "addr:street_address": "UNDEFINED"
fastrac_cafe_us.geojson has "addr:street_address": "Undefined"

I will see about killing off "undefined" from street numbers as well in this PR; tackle the others in other pipelines.

File: locations/pipelines/address_clean_up.py [hunk 1]
```diff
@@ -16,8 +16,14 @@ def merge_address_lines(address_lines: [str]) -> str:
 def clean_address(address: list[str] | str) -> str:
     if not address:
         return ""
+
+    if isinstance(address, str):
+        if address.lower() == "undefined":
```
Comment (iandees) [inline locations/pipelines/address_clean_up.py:21]: Maybe include a step to remove whitespace?

```suggestion
        if address.strip().lower() == "undefined":
```

**Comment by CloCkWeRX** [comment]
@iandees this one should be good to go

**Comment by iandees** [comment]
Thanks!

**Pull request merged**